### PR TITLE
refactor: use new `@metamask/keyring-api` layout (split packages)

### PR DESCRIPTION
**Pull request by ccharly**
## **Description**

Bumping accounts related dependencies + use the new `keyring-api` layout.

Those tests requires a higher timeout (which might not be true once merged to `main`)

`flags = {"circleci": {"timeoutMinutes": 35}}`

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28861?quickstart=1)

## **Related issues**

N/A

## **Manual testing steps**

N/A

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by zone-live — DISMISSED** [review]

**Comment by HowardBraham** [comment]
@metamaskbot update-policies

File: ui/components/multichain/account-list-menu/account-list-menu.tsx [hunk 1]
```diff
@@ -19,11 +19,11 @@ import {
   EthAccountType,
   SolAccountType,
   ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
-  InternalAccount,
   KeyringAccountType,
   ///: END:ONLY_INCLUDE_IF
 } from '@metamask/keyring-api';
 ///: BEGIN:ONLY_INCLUDE_IF(build-flask)
+import { InternalAccount } from '@metamask/keyring-internal-api';
```
Comment (danroc) [inline ui/components/multichain/account-list-menu/account-list-menu.tsx:26]: Check if it needs to be in the `build-flask` target

**Review by darioAnongba — DISMISSED** [review]

**Review by danroc — DISMISSED** [review]

**Review by jpuri — DISMISSED** [review]
changes in files owned by confirmations team look good 👍

**Review by Prithpal-Sooriya — DISMISSED** [review]
Approved from notifications side. It was just a type change.

**Review by matthewwalsh0 — DISMISSED** [review]

**Review by NidhiKJha — DISMISSED** [review]

**Review by darioAnongba — APPROVED** [review]

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by danroc — APPROVED** [review]

**Review by matthewwalsh0 — APPROVED** [review]

**Review by jonybur — APPROVED** [review]

**Pull request merged**